File: app/Models/ServiceCategory.php
<?php

namespace App\Models;

use App\Enums\BusinessStatus;
use App\Models\Concerns\BelongsToTenant;
use Illuminate\Database\Eloquent\Concerns\HasUlids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class ServiceCategory extends Model
{
    use BelongsToTenant, HasUlids, SoftDeletes;

    protected $fillable = ['branch_id', 'name', 'slug', 'description', 'image', 'status', 'sort_order'];

    protected function casts(): array
    {
        return ['status' => BusinessStatus::class];
    }

    public function branch(): BelongsTo
    {
        return $this->belongsTo(Branch::class);
    }

    public function services(): HasMany
    {
        return $this->hasMany(Service::class, 'category_id');
    }
}

File: mobile/lib/features/services/data/models/service_category.dart
/// Mirrors `App\Http\Resources\ServiceCategoryResource`.
class ServiceCategory {
  const ServiceCategory({
    required this.id,
    required this.branchId,
    required this.name,
    required this.slug,
    this.description,
    this.image,
    required this.status,
    required this.sortOrder,
  });

  factory ServiceCategory.fromJson(Map<String, dynamic> json) => ServiceCategory(
    id: json['id'] as String,
    branchId: json['branch_id'] as String,
    name: json['name'] as String,
    slug: json['slug'] as String,
    description: json['description'] as String?,
    image: json['image'] as String?,
    status: json['status'] as String,
    sortOrder: json['sort_order'] as int? ?? 0,
  );

  final String id;
  final String branchId;
  final String name;
  final String slug;
  final String? description;
  final String? image;
  final String status;
  final int sortOrder;
}
